src/NServiceBus.Metrics/MetricsConfigurationExtensions.cs

```diff
@@ -19,13 +19,8 @@ public static MetricsOptions EnableMetrics(this EndpointConfiguration endpointCo
             Guard.AgainstNull(nameof(endpointConfiguration), endpointConfiguration);
 
             var settings = endpointConfiguration.GetSettings();
-            var options = settings.GetOrCreate<MetricsOptions>();
-            settings.Set(typeof(MetricsFeature).FullName, FeatureState.Enabled);
-
-            endpointConfiguration.Recoverability().Immediate(c => c.OnMessageBeingRetried((m, ct) => options.Immediate(m, ct)));
-            endpointConfiguration.Recoverability().Delayed(c => c.OnMessageBeingRetried((m, ct) => options.Delayed(m, ct)));
-
-            return options;
+            settings.EnableFeatureByDefault<MetricsFeature>();
```

Comment:
Is there a subtle difference to settings.Set(typeof(MetricsFeature).FullName, FeatureState.Enabled); ?

Reply:
The way I remember it:

`settings.EnableFeatureByDefault<MetricsFeature>();` => on by default but if a user has disabled it or one of the prereqs are not true it won't be enabled

`settings.Set(typeof(MetricsFeature).FullName, FeatureState.Enabled);` => force it enabled no matter what

Reply:
@andreasohlund under that light I think `settings.EnableFeatureByDefault<MetricsFeature>();` is the better choice, no?

Reply:
Agree